specification/sbi_atee.adoc

```diff
@@ -10,8 +39,1296 @@ shown in the figure above, the TEE Runtime (operating in VS-mode) supports
 applications as TEEs when the host environment is a baremetal OS (not 
 running a VMM).
 
-The following operations must be implemented by the TSM ( _Formal input 
-parameters and return codes are TBD_ ).
+== Background
+Virtualization platforms are typically comprised of several components including
+platform firmware, host OS, VMM, and the actual payloads that run on them (typically
+in a VM). This model is well established, but the downside is that most platform
+components are in the TCB. This aspect is ill-suited for confidential compute workloads
+that rely on Trusted Execution Environments, and strive to minimize the TCB footprint.
+
+We seek to alleviate this concern by introducing the notion of the a trusted broker for
+TEEs known as the TEE Security Manger (TSM). The idea is that the TSM has a minimal possible
+footprint, and acts as a trusted intermediary between the the untrusted platform components,
+and the TEE workloads that run on it. The TCB (which includes the TSM and HW) enforces strict confidentiality
+and integrity security properties for code execution and data integrity for confidential workloads.
+It also isolates confidential workloads from all other platform components (non-confidential and confidential).
+The exact hardware and software mechanisms used to guarantee the isolation are platform dependent, but confidential
+workloads have execution, data integrity, and confidentiality protections from any other platform component. The
+responsibility of the TSM is to enforce the guarantees accorded to TEE workloads. Specifically, security invariants for
+confidential workloads are enforced by the TSM. The VMM manages security for non-confidential workloads, and resource and
+scheduling management functions for all workloads (confidential and non-confidential).
+
+In this scheme, compute resources like memory start off as traditional untrusted resources
+in the non-confidential world, and are transtioned to their confidential analogue via the TSM.
+Once the conversion process is complete, it can be assigned to a TVM via the TSM. A converted confidential-resource
+can be freely assigned to another TVM when it's no longer in use. However, an unused confidential-resource must be
+explicitly reclaimed for use in the non-confidential world (this is tracked and enforced by the TSM).
+
+The TEE address space can be comprised of confidential and non-confidential regions. The former includes
+both measured pages (that are part of the initial TVM payload), and confidential zero-pages that can be
+mapped-in on demand by the VMM following runtime access by the TVM. The non-confidential TVM-defined regions
+include those for shared-pages and MMIO.
+
```

Comment:
we should note here MMIO is non-confidential with no direct IO. once direct IO is supported - MMIO page mappings will also be allowed to be in the confidential memory region (simply since the mappings are in confidential memory).